test/functorch/test_control_flow.py

```diff
@@ -1744,6 +1761,102 @@ def fct_different_output_tuple(x, y):
         )
         self.assertEqual(result_diff, expected_result)
         self.assertEqual(result_diff[1], result_same[1][1])
+        
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise
+    # as the expand operation is non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+        ),
+    )
+    def test_associative_scan_expand(self, compile_mode, combine_mode, reverse, device):
+        x = torch.randn(3, 2, 2, device=device)
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+        
+        def body(x, y):
+            return x * torch.sum(y, -1).expand(x.shape)
+
+        result = scan_fct(
+            body,
+            x,
+            0,
+            reverse=reverse,
+            combine_mode=combine_mode,
+        )
+        expected_result = _fake_associative_scan(
+            body, x, 0, reverse=reverse
+        )
+        self.assertEqual(result, expected_result)
+        
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise
+    # as the expand operation is non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+        ),
+    )
+    def test_associative_scan_sparse_tensor(self, compile_mode, combine_mode, reverse, device):
```

Comment:
We don't need that many parametrized tests for failures. Can remove all the decorators?

Reply:
For the failure testcases, I removed the additional parameters